doc: create centralized documentation for developers (#1120)
HeyPuter/puter: doc/development/arch.md

# System Architecture

## General

> Puter's system goals is to provide an internet operating system, providing (as brief):

Reviewer: ```suggestion
Puter is a high-level distributed operating system, providing (as brief):
```

Rather than "Puter's system goals [are] to provide..." it would be more correct to say "Puter is...", i.e. "Puter is an internet operating system".

Although, in current documentation we want to use exactly this phrase:
> Puter is a High-Level Distributed Operating System

I've stopped writing it as "internet operating system" because that implies Puter to not be suitable for intranet use (i.e. LAN), which is not accurate since Puter can be used on any type of network in this regard.
